Resync WPT streams tests up to 1d3fbb8

youennf opened the pull request:
#### 361229b252e810422a2c47e9f8fdc8f21ca59f65
<pre>
Resync WPT streams tests up to 1d3fbb8
<a href="https://bugs.webkit.org/show_bug.cgi?id=299273">https://bugs.webkit.org/show_bug.cgi?id=299273</a>

Reviewed by Chris Dumez.

* LayoutTests/TestExpectations: Skip new timing out tests.
* LayoutTests/imported/w3c/web-platform-tests/streams: Resynced.
* LayoutTests/platform/mac-wk1/TestExpectations: Skip unsupported service worker/shared worker tests.
* Source/WebCore/Modules/streams/WritableStreamInternals.js:
(writableStreamAbort): We add a check, following the spec, to ensure that, in writer abort, we proceed if signaling the abortion did not already change the writer state.
See <a href="https://streams.spec.whatwg.org/#writable-stream-abort">https://streams.spec.whatwg.org/#writable-stream-abort</a> step 4.

Canonical link: <a href="https://commits.webkit.org/300370@main">https://commits.webkit.org/300370@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/4610301ae642957af671d176e6e4df0cb501f061

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/122154 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/41857 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/32526 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/128727 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/74248 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/536f46ce-ed8c-4eeb-a854-aca59ecdab52) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/124030 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/42571 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/50450 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/92845 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/61722 "Passed tests") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/b28c78fc-8f28-4e58-b9a9-ce1284a32684) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/125106 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/33953 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/109385 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/73499 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/efed8738-7e01-479f-bd3b-03216d8db195) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/32966 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/27550 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/72215 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/103461 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/27741 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/131478 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/49093 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/37347 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/101415 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/49467 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/105598 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/101284 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/25715 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/46656 "Passed tests") | [❌ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/24766 "Exiting early after 10 failures. 1589 tests run. 2 flakes") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/45844 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/48950 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/54684 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/48420 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/51770 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/50100 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/7e139d649e3d3c9a959a475105a6ae537d75c20a. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/51104

youennf committed:
Resync WPT streams tests up to 1d3fbb8
https://bugs.webkit.org/show_bug.cgi?id=299273

Reviewed by Chris Dumez.

* LayoutTests/TestExpectations: Skip new timing out tests.
* LayoutTests/imported/w3c/web-platform-tests/streams: Resynced.
* LayoutTests/platform/mac-wk1/TestExpectations: Skip unsupported service worker/shared worker tests.
* Source/WebCore/Modules/streams/WritableStreamInternals.js:
(writableStreamAbort): We add a check, following the spec, to ensure that, in writer abort, we proceed if signaling the abortion did not already change the writer state.
See https://streams.spec.whatwg.org/#writable-stream-abort step 4.

Canonical link: https://commits.webkit.org/300370@main
